cluster/kind/kind.go

```diff
@@ -0,0 +1,172 @@
+package kind
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/openconfig/kne/exec/run"
+	"golang.org/x/oauth2/google"
+	log "k8s.io/klog/v2"
+)
+
+const (
+	dockerConfigEnvVar = "DOCKER_CONFIG"
+	kubeletConfigPath  = "/var/lib/kubelet/config.json"
+)
+
+var (
+	kubeletConfigPathTemplate    = "%s:" + kubeletConfigPath
+	googleFindDefaultCredentials = google.FindDefaultCredentials
+	clusterKindNameRE            = regexp.MustCompile("kind-(.*)")
+)
+
+func ClusterIsKind() (bool, error) {
```

Comment:
See code dup in clusterKindName(). Can we call clusterKindName() to tell whether it is a kind cluster?

Reply:
I considered this, but clusterKindName treats a valid non-kind cluster as an error, whereas ClusterIsKind only errors on kubectl exec error. So technically this small dup is required